pkg/crd/schema.go

```diff
@@ -341,7 +341,7 @@ func structToSchema(ctx *schemaContext, structType *ast.StructType) *apiext.JSON
 		Properties: make(map[string]apiext.JSONSchemaProps),
 	}
 
-	if ctx.info.RawSpec.Type != structType {
+	if ctx.info.RawSpec != nil && ctx.info.RawSpec.Type != structType {
```

Comment:
could we add a test case?

Reply:
Hi, I've managed to break the complex set of structs down to the following test case. `controller-get crd
paths="<path/to/testcase/below>/..." output:stdout` currently fails on this one. With the fix it doesn't fail. It's due to the use of an anonymous struct.

```
// +groupName=test
package config

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

type ExampleSpec struct {
	Service Service `json:"service"`
}

type Example struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec ExampleSpec `json:"spec,omitempty"`
}

type Service struct {

	// +optional
	Foobar struct {
		// placeholder
	} `yaml:"foobar" json:"foobar"`
}
```

Reply:
@rkettelerij please update pkg/crd/testdata/cronjob_types.go with an anonymous struct to test this

/ok-to-test